### PR TITLE
Small adjustments to fabricbot automation

### PR DESCRIPTION
**Pull request by joperezr**
Small fixes to the automation configuration

- Removed a couple of conflicting search terms that would have caused a task not to run
- Removed a couple of tasks that were just disabled.
- Removing auto-merge feature in favor of using GH's version instead.
- Keeping milestones on issues after they get closed in order to track which release was an issue fixed in.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/3988)

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]

**Review by danmoseley — APPROVED** [review]

**Review by danmoseley — APPROVED** [review]

File: .github/fabricbot.json
Comment (RussKie) [inline .github/fabricbot.json:937]: Why remove this?

**Review by RussKie — APPROVED** [review]
LGTM modulo the question

File: .github/fabricbot.json
Comment (RussKie) [reply .github/fabricbot.json:937]: The way that I eventually settled on in other repos was to milestone pull-requests - since those represent the code changes that went into a specific milestone. Issues automatically get closed by GitHub itself (assuming PRs use the keywords), and there's no real way for the bot to know which milestone the issue belongs to. Besides, an issue could be resolved by multiple PRs.
An issue at triage time may be tagged with a milestone, but in my experience that's always speculative and, in reality, an issue may get closed manually (e.g., decided its not worth the effort) or resolved by a change in a completely different milestone. So to avoid any possible confusion this step demilestones closed issues.

Hope this makes sense. And I'm happy to discuss this topic further, may be there's some other angle that I miss.

**Comment by danmoseley** [comment]
should we delete one or both of these automerge labels?
https://github.com/dotnet/extensions/labels?q=auto

**Comment by RussKie** [comment]
> should we delete one or both of these automerge labels? [dotnet/extensions/labels (auto)](https://github.com/dotnet/extensions/labels?q=auto)

Yes, once this is merged.

**Pull request closed without merging**
